component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -758,28 +758,23 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
             }
             throw new InvalidCredentialsException("Retrying to resend the OTP");
         }
-
+        boolean succeededAttempt = false;
         if (userToken.equals(contextToken)) {
             processValidUserToken(context, authenticatedUser);
+            succeededAttempt = true;
         } else if (isLocalUser && "true".equals(SMSOTPUtils.getBackupCode(context))) {
-            checkWithBackUpCodes(context, userToken, authenticatedUser);
+            succeededAttempt = checkWithBackUpCodes(context, userToken, authenticatedUser);
         } else {
+            context.setProperty(SMSOTPConstants.CODE_MISMATCH, true);
```

Comment:
Can we add the debug log here?

Reply:
Fixed in https://github.com/wso2-extensions/identity-outbound-auth-sms-otp/pull/94/commits/a8b29a5973113ffb46c61ce362ed50d6877eb259